l10n_eu_nace/wizard/res_partner_industry_eu_nace_wizard.py

```diff
@@ -0,0 +1,183 @@
+            PREFIX xkos: <http://rdf-vocabulary.ddialliance.org/xkos#>
+            PREFIX xml: <http://www.w3.org/XML/1998/namespace>
+            PREFIX xsd: <http://www.w3.org/2001/XMLSchema#>
+
+            SELECT ?LEVEL ?code ?parentCode {select_query} WHERE {{
+                ?s skos:inScheme :nace2 ;
+                {skos_query}
+                skos:notation ?code;
+                skos:broader ?parent;
+                ^skos:member ?Member.
+                ?parent skos:notation ?parentCode .
+                {filter_query}
+                FILTER (?Member = :sections || ?Member = :divisions  ||
+                ?Member = :groups || ?Member = :classes)
+                BIND (STR(?Member) AS ?NACELEVEL)
+                BIND (STRAFTER(?NACELEVEL, "/nace2/") AS ?LEVEL)
+                ?Member skos:prefLabel ?MemberLabel .
+                FILTER (LANG(?MemberLabel) = "en")
+            }} ORDER BY ?code
+        """
+        return query
+
+    def _create_nace_industry(self, nace_data, languages):
+        """Creates the partner industry records from the NACE data and adds it's
+        translations."""
+        nace_ids = self.env["res.partner.industry"]
+        nace_json = nace_data.json()
+        bindings = nace_json.get("results", {}).get("bindings", {})
+        for binding in bindings:
+            nace_code = binding.get("code", {}).get("value", "")
+            nace_id = self.env["res.partner.industry"].search(
+                [("full_name", "=like", nace_code + "%")], limit=1
+            )
+            nace_parent_code = binding.get("parentCode", {}).get("value", "")
+            parent_id = self.env["res.partner.industry"].search(
+                [("full_name", "=like", nace_parent_code + "%")], limit=1
+            )
+            nace_name = binding.get("EN", {}).get("value")
+            if not nace_id:
+                nace_id = self.env["res.partner.industry"].create(
+                    {
+                        "name": nace_name,
+                        "full_name": nace_name,
+                        "parent_id": parent_id and parent_id.id,
+                    }
+                )
+                nace_ids |= nace_id
+            for lang, lang_code in languages:
+                nace_translated = binding.get(lang_code, {}).get("value")
+                nace_id.with_context(lang=lang).write(
+                    {
+                        "name": nace_translated,
+                        "full_name": nace_translated,
+                    }
+                )
+        return nace_ids
+
+    def update_partner_industry_eu_nace(self):
+        languages = self.get_languages()
+        query = self._create_query(languages)
+        result = requests.get(
+            ENDPOINT, params={"format": "json", "query": query}, timeout=60
+        )
+        result.raise_for_status()
+        nace_ids = self._create_nace_industry(result, languages)
+        return nace_ids
```

Comment:
I think you should return action with created naces.

Reply:
Done!